### PR TITLE
chore: add create indexes section to notebook

### PR DESCRIPTION
**Pull request by eeshan9815**
## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file) N/A, since documentation is added in Notebook itself.
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] These samples need a new **API enabled** in testing projects to pass (let us know which ones) N/A APIs are enabled via Colab notebook when they are run.
- [x] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones) N/A
- [x] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [x] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by jackwotherspoon** [comment]
@eeshan9815 do you mind updating the PR by copying over only the newly created cells and adding them to the markdown in the correct location? This will better preserve the history here and allow us to review only the new code cells.

File: cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb [hunk 1]
```diff
@@ -639,6 +639,128 @@
...
+    },
+    "execution_count": null,
+    "outputs": []
+  },
+  {
+    "cell_type": "code",
+    "source": [
+      "# @markdown Create an IVFFLAT index on the `product_embeddings` table:\n",
+      "lists =  100 # @param {type:\"integer\"}\n",
+      "operator =  \"vector_cosine_ops\"  # @param [\"vector_cosine_ops\", \"vector_l2_ops\", \"vector_ip_ops\"]\n",
```
Comment (jackwotherspoon) [inline cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb:718]: is there any value to defining operator here in the exact same way as the previous cell? i.e. if I had set `operator = "vector_cosine_ops"` in the previous cell is there a use-case where I would want to switch it in this cell? If not then lets remove it.

File: cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb [hunk 2]
```diff
@@ -639,6 +639,128 @@
...
+      "id": "Ley1ZGjnG5Fr"
+    },
+    "source": [
+      "### Create indexes for faster similarity search in pgvector\n",
+      "\n",
+      "- Vector indexes can significantly speed up similarity search operation and avoid the brute-force exact nearest neighbor search that is used by default.\n",
+      "\n",
+      "- pgvector comes with two types of indexes (as of v0.5.1): `hnsw` and `ivfflat`.\n",
+      "\n",
+      "> 💡 Click [here](https://cloud.google.com/blog/products/databases/faster-similarity-search-performance-with-pgvector-indexes) to learn more about pgvector indexes.\n",
```
[the remark this answers is not in the document]
Comment (eeshan9815) [reply cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb:654]: Yes, the blog has just been released. The link works now.

File: cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb [hunk 3]
```diff
@@ -639,6 +639,128 @@
...
+      "\n",
+      "Enter or modify the values of index parameters for your index of choice and run the corresponding cell:"
+    ]
+  },
+  {
+    "cell_type": "code",
+    "source": [
+      "# @markdown Create an HNSW index on the `product_embeddings` table:\n",
+      "m =  24 # @param {type:\"integer\"}\n",
```
[the remark this answers is not in the document]
Comment (eeshan9815) [reply cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb:663]: Yes, `m` is the standard name for this HNSW index parameter. It is desribed in greater detail in the accompanying blog as well: https://cloud.google.com/blog/products/databases/faster-similarity-search-performance-with-pgvector-indexes. The same name goes on the official pgvector site too: https://github.com/pgvector/pgvector.

File: cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb (hunk 1 above)
Comment (eeshan9815) [reply cloud-sql/postgres/pgvector/notebooks/pgvector_gen_ai_demo.ipynb:718]: Yes, these are two types of indexes, and each of them could have their own `operator` defined based on the requirement.

**Review by jackwotherspoon — APPROVED** [review]
LGTM

**Pull request merged**